### PR TITLE
Answer FAQ

### PR DESCRIPTION
**Pull request by osterman**
## what
* Address https://github.com/cloudposse/docs/issues/161
* Address https://github.com/cloudposse/docs/issues/160
* Address https://github.com/cloudposse/docs/issues/146
* Address https://github.com/cloudposse/docs/issues/131
* Address https://github.com/cloudposse/docs/issues/157

### CODE REVIEW

File: content/faq/kiam-error-detecting-arn-prefix.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+title: "Kiam: error detecting arn prefix: aws metadata api not available"
```
Comment (aknysh) [inline content/faq/kiam-error-detecting-arn-prefix.md]: Capitalize
Kiam: error detecting ARN prefix: AWS metadata API not available

File: content/faq/kube2iam-assuming-role-access-denied.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+```
+time="2018-06-19T19:58:43Z" level=error msg="Error assuming role AccessDenied: User: arn:aws:sts::957237095956:assumed-role/nodes.us-west-2.staging.example.net/i-007b0d693468102fb is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::957237095956:role/example-staging-external-dns\n\tstatus code: 403, request id: 2b14d0eb-73fb-11e8-8aaa-578a2a66ee2c" ns.name=kube-system pod.iam.role="arn:aws:iam::957237095956:role/example-staging-external-dns" req.method=GET req.path=/latest/meta-data/iam/security-credentials/example-staging-external-dns req.remote=100.96.162.129
+```
+
+# Answer
+
+We've seen this error with `kube2iam` many times due to AWS API rate limiting. It affects even small kubernetes clusters running `kube2iam`. For AWS, the request rate limit is account-wide and other applications deployed in the same account may be depleting the budget independently. There are fundamental [problems with the architecutre of `kube2iam`](https://medium.com/@pingles/kiam-iterating-for-security-and-reliability-5e793ab93ec3) that are addressed in [`kiam`](https://github.com/uswitch/kiam).
```
Comment (aknysh) [inline content/faq/kube2iam-assuming-role-access-denied.md]: capitalize Kubernetes

File: content/faq/terraform-value-of-count-cannot-be-computed.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+```
+
+# Answer
+
+Terraform has well-known issues with `counts` in `maps`.
+
+Terraform (in the current incarnation) is very sensitive to these two things:
+
+1. Dynamic `counts` across modules -- when you have a dynamic count (calculated by some expression with input params) in one module and then use the module from other modules
+2. It does not especially like those dynamic `counts` in `maps` and `lists`
```
Comment (aknysh) [inline content/faq/terraform-value-of-count-cannot-be-computed.md]: It does not especially like dynamic `counts` with `maps` and `lists`

File: content/faq/terraform-value-of-count-cannot-be-computed.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+> - <https://github.com/hashicorp/terraform/issues/12570>
+> - <https://github.com/hashicorp/terraform/issues/17048>
+
+This issue has come up for us time and again. If a module has a map variable and has interpolation inside this map variable, count inside a module results in value of `count` that cannot be computed. It appears that this error only occurs when running `terraform apply` on a new environment, but not any existing environment! So the problem often goes unnoticed.
+
+Here's an example from one of our modules.
+
+Here the `count` depends on the `map` and the input `var.tags`:
+
+> - <https://github.com/cloudposse/terraform-null-label/blob/master/main.tf#L23>
```
Comment (aknysh) [inline content/faq/terraform-value-of-count-cannot-be-computed.md]: line numbers might change

File: content/faq/terraform-value-of-count-cannot-be-computed.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+And here `var.tags` depends on the `map`, the other inputs and on the `data` provider:
+
+> - <https://github.com/cloudposse/terraform-aws-ec2-instance/blob/master/main.tf#L68a>
```
Comment (aknysh) [inline content/faq/terraform-value-of-count-cannot-be-computed.md]: line numbers might change

File: content/geodesic/kops/upgrade-cluster.md [hunk 6]
```diff
@@ -13,23 +13,25 @@ RUN make -C /packages/install kops KOPS_VERSION=1.9.0
 
 Find the latest release of kops [here](https://github.com/kubernetes/kops/releases).
 
-Then follow the instructions for upgrading kubernetes.
+Then follow the instructions for upgrading kubernetes below.
```
Comment (aknysh) [inline content/geodesic/kops/upgrade-cluster.md]: Then follow the instructions below for upgrading Kubernetes.

File: content/geodesic/kops/upgrade-cluster.md [hunk 7]
```diff
@@ -13,23 +13,25 @@ RUN make -C /packages/install kops KOPS_VERSION=1.9.0
...
 
 {{% dialog type="info" icon="fa fa-book" title="Read More" %}}
 <https://github.com/kubernetes/kops/blob/master/docs/releases.md>
 {{% /dialog %}}
 
 # Upgrading Kubernetes Release
 
-Upgrading `kops` cluster to a new release of kubernetes requires that we upgrade the `kubectl` client as well as select a kubernetes release that is compatible with kops.
+Upgrading `kops` cluster to a new release of kubernetes requires that we upgrade the `kubectl` client in `geodesic` as well as select a kubernetes release that is compatible with kops. Note, `kops` usually lags behind in support for the latest release of Kubernetes. If the latest release of `kops` is `v1.9`, then that means it supports up to kubernetes `v1.9` and will not be compatible with `v1.10` or newer.
```
Comment (aknysh) [inline content/geodesic/kops/upgrade-cluster.md]: let's capitalize Kubernetes and use it consistently

File: content/geodesic/kops/upgrade-cluster.md [hunk 8]
```diff
@@ -13,23 +13,25 @@ RUN make -C /packages/install kops KOPS_VERSION=1.9.0
...
 
 To explictly upgrade `kubectl`, add this to the `Dockerfile`
+
 ```
 RUN make -C /packages/install kubectl KUBECTL_VERSION=1.10.0
 ```
- The `kubectl` release version corresponds to the kubernetes release. Find the latest release of kubernetes [here](https://github.com/kubernetes/kubernetes/releases).
 
- Then follow the [official instructions](https://github.com/kubernetes/kops/blob/master/docs/upgrade.md) for upgrading kops.
+The `kubectl` release version corresponds to the kubernetes release. Find the latest release of kubernetes [here](https://github.com/kubernetes/kubernetes/releases).
```
Comment (aknysh) [inline content/geodesic/kops/upgrade-cluster.md]: let's capitalize Kubernetes and use it consistently

File: content/geodesic/kops/upgrade-cluster.md [hunk 9]
```diff
@@ -57,21 +59,41 @@ Then [rebuild the manifest and update the cluster]({{< relref "geodesic/kops/man
 {{% /dialog %}}
 
 
-# Update manifest
+# Update Manifest
+
+The [`build-kops-manifest`](https://github.com/cloudposse/geodesic/blob/master/rootfs/usr/local/bin/build-kops-manifest) command will regenerate the `/conf/kops/manifest.yaml`.
```
Comment (aknysh) [inline content/geodesic/kops/upgrade-cluster.md]: will regenerate the `/conf/kops/manifest.yaml` file.

File: content/tools/helm.md [hunk 10]
```diff
@@ -13,7 +13,15 @@ Helm makes it easy to install `charts` (an application) on kubernetes clusters.
 
 The helm `tiller` is the server-side component of Helm. It's used to cordinate with the kubernetes cluster in order to deploy resources (E.g. `Services`, `Deployments`, `StatefulSets`, etc.)
 
-To install the `tiller`, just run `helm init` from the Geodesic Shell after assumming-role and running `kops export kubecfg $CLUSTER_NAME`.
+To install the `tiller`, just run `helm init` from the Geodesic Shell after assumming-role and running `kops export kubecfg`.
```
Comment (aknysh) [inline content/tools/helm.md]: let's not capitalize `shell`

File: content/geodesic/kops/upgrade-cluster.md (hunk 7 above)
Comment (osterman) [reply content/geodesic/kops/upgrade-cluster.md]: https://github.com/cloudposse/docs/issues/179

File: content/geodesic/kops/upgrade-cluster.md (hunk 8 above)
Comment (osterman) [reply content/geodesic/kops/upgrade-cluster.md]: https://github.com/cloudposse/docs/issues/179

**Pull request merged**